src/iterative_sorting/iterative_sorting.py

```diff
@@ -1,30 +1,44 @@
-# TO-DO: Complete the selection_sort() function below 
-def selection_sort( arr ):
+# TO-DO: Complete the selection_sort() function below
+def selection_sort(arr):
     # loop through n-1 elements
     for i in range(0, len(arr) - 1):
         cur_index = i
         smallest_index = cur_index
         # TO-DO: find next smallest element
-        # (hint, can do in 3 loc) 
-             
-
-
+        # (hint, can do in 3 loc)
+        while cur_index < len(arr):
+            if arr[cur_index] < arr[smallest_index]:
+                smallest_index = cur_index
+            cur_index += 1
 
         # TO-DO: swap
-
-
-
+        arr[i], arr[smallest_index] = arr[smallest_index], arr[i]
 
     return arr
 
 
 # TO-DO:  implement the Bubble Sort function below
-def bubble_sort( arr ):
-
+def bubble_sort(arr):
+    def sort_loop():
+        cur_index = 0
+        swaps = 0
+        # Loop through your array
+        while cur_index < len(arr) - 1:
```

Comment:
Solid implementation. Again, `for i in range(0, len(arr) - 1` will achieve the same thing but its a matter of preference.